controlplane/pkg/model/base.go

```diff
@@ -57,28 +57,35 @@ func (b *baseDomain) store(key string, value cloneable) {
 func (b *baseDomain) delete(key string) {
 	b.mtx.Lock()
 	defer b.mtx.Unlock()
+
 	v, exist := b.innerMap[key]
 	if !exist {
 		return
 	}
+
 	delete(b.innerMap, key)
 	b.resourceDeleted(v)
 }
 
-func (b *baseDomain) applyChanges(key string, changeFunc func(interface{})) interface{} {
+func (b *baseDomain) applyChanges(key string, value cloneable, changeFunc func(interface{})) interface{} {
 	b.mtx.Lock()
 	defer b.mtx.Unlock()
 
 	old, exist := b.innerMap[key]
 	if !exist {
-		return nil
+		old = value
```

Comment:
I think it is a bit unobvious for customers of ApplyClientConnectionChanges method. Passed pointer to structure will be correctly updated even if there is no more such entity in the model. Probably it makes sense to return some boolean flag? What do you think?

Reply:
I have changed `Add*` [1], `Delete*` [2], `Apply*Changes` [3] methods to return `error` if:
1. `key` exists in domain
2, 3. `key` doesn't exist in domain
5c5ca2f